java-checks/src/main/java/org/sonar/java/checks/FunctionalInterfacesSpecialisedCheck.java

```diff
@@ -0,0 +1,257 @@
+      case "java.util.function.Predicate":
+        reportString = handleOneParameterInterface(args);
+        return reportString != null ? (reportString + "Predicate") : null;
+      case "java.util.function.UnaryOperator":
+        reportString = handleOneParameterInterface(args);
+        return reportString != null ? (reportString + "UnaryOperator") : null;
+      case "java.util.function.BinaryOperator":
+        reportString = handleOneParameterInterface(args);
+        return reportString != null ? (reportString + "BinaryOperator") : null;
+      default:
+        return null;
+    }
+  }
+
+  @CheckForNull
+  private static String handleOneParameterInterface(TypeArguments args) {
+    return new ParameterTypeNameAndTreeType(args.get(0)).paramTypeName;
+  }
+
+  @CheckForNull
+  private static String handleFunctionInterface(TypeArguments args) {
+    ParameterTypeNameAndTreeType firstArgument = new ParameterTypeNameAndTreeType(args.get(0));
+    ParameterTypeNameAndTreeType secondArgument = new ParameterTypeNameAndTreeType(args.get(1));
+
+    if (firstArgument.paramTypeName != null && secondArgument.paramTypeName != null) {
+      if (firstArgument.paramTypeName.equals(secondArgument.paramTypeName)) {
+        return (UNARY_OPERATOR + firstArgument.originalName + ">");
+      }
+      return firstArgument.paramTypeName + "To" + secondArgument.paramTypeName + "Function";
+    } else if (firstArgument.paramTypeName == null && secondArgument.paramTypeName != null) {
+      return "To" + secondArgument.paramTypeName + "Function<" + firstArgument.originalName + ">";
+    } else if (firstArgument.paramTypeName != null) {
+      return firstArgument.paramTypeName + "Function<" + secondArgument.originalName + ">";
+    } else if (firstArgument.originalName.equals(secondArgument.originalName)) {
+      return !args.get(0).is(Tree.Kind.PARAMETERIZED_TYPE) ? (UNARY_OPERATOR + firstArgument.originalName + ">") : (UNARY_OPERATOR + firstArgument.originalName);
+    }
+    return null;
+  }
+
+  @CheckForNull
+  private static String handleBiFunctionInterface(TypeArguments args) {
+    ParameterTypeNameAndTreeType firstArgument = new ParameterTypeNameAndTreeType(args.get(0));
+    ParameterTypeNameAndTreeType secondArgument = new ParameterTypeNameAndTreeType(args.get(1));
+    ParameterTypeNameAndTreeType thirdArgument = new ParameterTypeNameAndTreeType(args.get(2));
+    return (firstArgument.originalName.equals(secondArgument.originalName) &&
+      firstArgument.originalName.equals(thirdArgument.originalName)) ? ("BinaryOperator<" + firstArgument.originalName + ">") : null;
+  }
+
+  @CheckForNull
+  private static String handleBiConsumerInterface(TypeArguments args) {
+    return "Obj" + returnStringFromJavaObject(((IdentifierTree) args.get(1)).symbolType()) + "Consumer";
+  }
+
+  @CheckForNull
+  private static String returnStringFromJavaObject(Type argType) {
+    if (argType.is("java.lang.Integer")) {
```

Comment:
As a good practice, I would prefer to have it the other way (same for others): https://next.sonarqube.com/sonarqube/coding_rules?open=squid%3AS1132&rule_key=squid%3AS1132